crates/wasmtime/src/runtime/component/func/options.rs

```diff
@@ -196,7 +196,7 @@ pub struct LowerContext<'a, T> {
     /// into.
     ///
     /// This pointer is required to be owned by the `store` provided.
-    instance: *mut ComponentInstance,
+    pub(crate) instance: *mut ComponentInstance,
```

Comment:
For this and the below change I had to reread and re-figure-out what I was originally intending with this. I think it would be preferable to not add `pub(crate)` here as that expands the number of locations that need to be audited when working with this field.

I think it should be ok though to add accessors like `fn instance(&self) -> &ComponentInstance` and `fn instance_mut(&mut self) -> &mut ComponentInstance` to these types (lift/lower contexts). That would remove the need for `unsafe` at access sites and might clean up some preexisting access sites as well